### DISCUSSION TITLE
Path Key not Available in safeParse

### DISCUSSION THREAD

**Discussion by jdgamble555** [open-circle/valibot#696] [General]
I'm trying to create an error message, but there doesn't seem to be any way to tell which `key` or input the problem is coming from, since the path variable is null. Here is the `issue`. I am passing an empty input object. 

I need to tell the user which input is the problem... ?
```ts
{
  kind: 'schema',
  type: 'string',
  input: undefined,
  expected: 'string',
  received: 'undefined',
  message: 'Invalid type: Expected string but received undefined',
  requirement: undefined,
  path: undefined,
  issues: undefined,
  lang: undefined,
  abortEarly: true,
  abortPipeEarly: undefined
}
```
with this simple schema:
```ts
export const courseSchema = object({
    name: string(),
    description: string(),
    visibility: string()
});
```

Thanks,

J

**Comment by fabian-hiller** [comment]
Can you reproduce the problem in our [playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6CBQBWfJjAAUAb3pxVcVshB8AXOI5RgrAObyAlABoVagCZ9UTA2BjA2ujGP2GTFq6ozBUSmByGABPNw8YA2MzegBfUwZmNg44KDsAV1J4UXdUZGw+AAVkKFQ+eW5+QXM4ZVUNLV0M1ltsQz5rS1Vbe0dnVyJCbswAoJDwocsEpJZ2CFI+MVIIE3TULJhEoA)?

**Reply by jdgamble555** [reply]
Can ignore this. I wasn't using `decode` or `Object.fromEntries()` from my form data.

Thanks,

J

**Reply by jdgamble555** [reply]
Actually yes, it still doesn't work when you try and change the message:

[Playground](https://valibot.dev/playground/?code=JYWwDg9gTgLgBAKjgQwM5wG5wGZQiOAcg2QBtgAjCGQgbgCh6BjCAO1XgGUmALAUxDI4AXkwA6CBQBWfJjAAUAb3pxVcVshB8AXOI5RgrAObyAlABoVagCZ9UTA2BjA2ujGP2GTFq6ozBUSmByGABPNw8YA2MzegBfUwZmNg44KDsAV1J4UXdUZGw+AAVkKFQ+eW5+QXM4ZVUNLV0M1ltsQz5rS1Vbe0dnVyJCbswAoJDwocs42vq1LVR8ox04eQDUDL5TEQA+Ot81VRZ2CFI+MVIIE3XNxIPD4GxVm-OwZBgebbnDn7hj1LeHwAkjABCI4C8xICeABtAAMAF0GL9fo9VtCQQIJAYjIYRMJRIQANZ8UKEL73FGHdIwDJQVhwAAGABJFBjQSAxCTQnE4ABaOCs9lYkikTa8gWsyELJZ8OKM5FUtRxSmHFVKtJ8Wn0iGLTZiGXIZaK1QqhIMIA)

J

**Reply by fabian-hiller** [reply]
Each schema is independent and therefore has no information about its parent. Because the parent lazily adds the path to the issue, it is not available when the child's message is calculated.

**Reply by jdgamble555** [reply]
Hey Fabian, you don't think it is important to be able to calculate a new message with the actual name of the input that has the problem? Granted, yes, this can be done locally instead of globally. Just a thought.

**Reply by fabian-hiller** [reply]
I agree that this functionality would add value to the library, but it would force us to compute the message at the very end when calling `parse` or `safeParse`.
